src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -278,7 +278,8 @@ public void databaseClosing() {
     protected void close() {
         // shutdown ScheduledExecutorService. Without shutdown, cause thread leak
         if (remoteRequestExecutor != null && !remoteRequestExecutor.isShutdown()) {
-            Utils.shutdownAndAwaitTermination(remoteRequestExecutor);
+            // Note: Time to wait is set 60 sec because RemoteRequest's socket timeout is set 60 seconds.
+            Utils.shutdownAndAwaitTermination(remoteRequestExecutor, 60);
         }
```

Comment:
http://developer.android.com/reference/org/apache/http/client/methods/AbortableHttpRequest.html
But HttpClient is not derived from that.

I saw this from the HttpClient.html you sent above:
```
This interface was deprecated in API level 22.
Please use openConnection() instead. Please visit this webpage for further details.
http://android-developers.blogspot.com/2011/09/androids-http-clients.html
```